shake256-33bytes-demo/methods/guest/src/main.rs

```diff
@@ -0,0 +1,130 @@
+#![no_std]
+#![no_main]
+extern crate alloc;
+
+use alloc::vec::Vec;
+use risc0_zkvm::guest::env;
+use serde::{Deserialize, Serialize};
+
+// ---------- module 1 ----------
+mod ser_bytes33 {
+    use core::fmt;
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
+    use serde::de::{self, SeqAccess, Visitor};
+
+    #[derive(Clone, Copy, PartialEq, Eq, Debug)]
+    pub struct Bytes33(pub [u8; 33]);
+
+    impl Serialize for Bytes33 {
+        fn serialize<S: Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
+            s.serialize_bytes(&self.0)
+        }
+    }
+    impl<'de> Deserialize<'de> for Bytes33 {
+        fn deserialize<D: Deserializer<'de>>(d: D) -> Result<Self, D::Error> {
+            struct V;
+            impl<'de> Visitor<'de> for V {
+                type Value = Bytes33;
+                fn expecting(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                    write!(f, "exactly 33 bytes")
+                }
+                fn visit_bytes<E: de::Error>(self, v: &[u8]) -> Result<Self::Value, E> {
+                    if v.len() != 33 { return Err(E::invalid_length(v.len(), &"33 bytes")); }
+                    let mut a = [0u8; 33];
+                    a.copy_from_slice(v);
+                    Ok(Bytes33(a))
+                }
+                fn visit_seq<A: SeqAccess<'de>>(self, mut seq: A) -> Result<Self::Value, A::Error> {
+                    let mut a = [0u8; 33];
+                    for i in 0..33 {
+                        a[i] = seq.next_element()?
+                            .ok_or_else(|| de::Error::invalid_length(i, &"33 bytes"))?;
+                    }
+                    Ok(Bytes33(a))
+                }
+            }
+            d.deserialize_bytes(V)
+        }
+    }
+    impl AsRef<[u8; 33]> for Bytes33 { fn as_ref(&self) -> &[u8; 33] { &self.0 } }
+    impl AsRef<[u8]> for Bytes33 { fn as_ref(&self) -> &[u8] { &self.0 } }
+}
+use ser_bytes33::Bytes33; // bring into crate scope
+// ---------- end module 1 ----------
+
+// ---------- module 2 ----------
+mod crypto {
+    #![allow(clippy::needless_borrows_for_generic_args)]
+
+    extern crate alloc;
+    use alloc::vec; 
+    use alloc::vec::Vec;
+    use sha2::{Digest, Sha256};
+    use tiny_keccak::{Hasher, Shake};
+
+    pub fn nssa_kdf(
+        ss_bytes: [u8; 32],
+        epk: &[u8; 33],
+        ipk: &[u8; 33],
+        commitment: &[u8; 32],
+        out_index: u32,
+    ) -> [u8; 32] {
+        let mut hasher = Sha256::new();
+        sha2::Digest::update(&mut hasher, b"NSSA/v0.1/KDF-SHA256");
```

Comment:
You can use update as method of `hasher`. The same for `sh` and other hashers in the code
```suggestion
        hasher.update( b"NSSA/v0.1/KDF-SHA256");
```